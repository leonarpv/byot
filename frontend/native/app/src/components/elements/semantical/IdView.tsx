import * as React from 'react';
import {View} from 'react-native';

interface Props {
  testID: string;
  children?: React.ReactNode;
}

export const IdView: React.FC<Props> = (props: Props) => {
  return (
    <View testID={props.testID} accessibilityLabel={props.testID}>
      {props.children}
    </View>
  );
};
