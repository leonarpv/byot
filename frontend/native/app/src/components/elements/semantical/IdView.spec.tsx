import * as React from 'react';
import {render} from '@testing-library/react-native';
import {IdView} from './IdView';

describe('<IdView/>', () => {
  it('should render', () => {
    const {asJSON} = render(<IdView testID="test-id" />);
    expect(asJSON()).toMatchSnapshot();
  });
});
