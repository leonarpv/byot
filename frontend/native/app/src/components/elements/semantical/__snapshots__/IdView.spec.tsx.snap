// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<IdView/> should render 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessibilityLabel="test-id"
    testID="test-id"
  />
</View>
`;
